Feature/ij 57 admin manage crisis

IeuanSkinner opened the pull request:
Team members can now see all of the Crisis Event records that have been created from their scope via the 'Crisis Events' button on the homepage. This will take them to the index view which lists only the active crisis events with the option of viewing the closed crisis events via a button which will take the user to a paginated view for all of the closed crisis events sorted by creation date newest to oldest.

A team member can click on a crisis event in either the active crisis events index or the paginated closed crisis events index and be taken to the crisis event member record where they can see the full details of the crisis events along with the option to close the crisis event and add notes to the crisis event.

Crisis events button on homepage
<img width="1440" alt="Screenshot 2021-03-16 at 14 30 35" src="https://user-images.githubusercontent.com/6815945/111327338-35f50100-8665-11eb-82e3-8b5fd9112606.png">

Active crisis events index
<img width="1440" alt="Screenshot 2021-03-16 at 14 30 48" src="https://user-images.githubusercontent.com/6815945/111327420-473e0d80-8665-11eb-8d4f-1a1935518f13.png">

Paginated closed crisis events index
<img width="1440" alt="Screenshot 2021-03-16 at 14 30 54" src="https://user-images.githubusercontent.com/6815945/111327491-5755ed00-8665-11eb-961a-e496fae61288.png">

Crisis event member view
<img width="1440" alt="Screenshot 2021-03-16 at 14 31 11" src="https://user-images.githubusercontent.com/6815945/111327570-663c9f80-8665-11eb-83c9-3d2f9ab85072.png">

phil-6 reviewed (changes requested):
Code looks good however some things don't appear to be working fully on the front end.

Closed Crisis Events page
![image](https://user-images.githubusercontent.com/29845460/111341264-44491a00-8671-11eb-9270-5f83a2afa321.png)
Could also do with a link back to active events.
Replace last update column with closed_at

Show Crisis Event
![image](https://user-images.githubusercontent.com/29845460/111341360-5925ad80-8671-11eb-8670-e4405f79c6e5.png)
Could do with a "back to list" button.
Change "close" to "close event"
If "Additional Info" is empty then display "User has not provided any additional info"
update time to include time since event (eg 35 minutes ago, or a line which says something like "Nadia Will hit the SOS button 35 minutes ago")


It would be nice to have an "edit note" button, (maybe time limited for 5 minutes) to allow of correcting of spelling mistakes / typos etc.

It would be good to have the "closed" at date displayed on closed crisis events.

Active crisis events page is now being weird:
![image](https://user-images.githubusercontent.com/29845460/111342012-f5e84b00-8671-11eb-89d3-5e608d8a03a3.png)
![image](https://user-images.githubusercontent.com/29845460/111342106-0a2c4800-8672-11eb-904a-93e93edcb649.png)

It would be good to update tables for mobile so their overflow is scrollable instead of expanding the entire viewport. I've done this on SP like this:
```scss  
table {
    overflow-x: auto;
    white-space: nowrap;
    display: block;
    @media (min-width: 576px) {
      display: table;
    }
    ...
}
```

On mobile remove x padding from crisis_event container, and x margin from row to make better use of limited horizontal space.

Notes don't work on mobile.
![image](https://user-images.githubusercontent.com/29845460/111343433-54fa8f80-8673-11eb-9c36-d12be3493ac8.png)

We also want to display the users phone number on the event info.

IeuanSkinner commented:
@phil-6 I believe I've addressed everything you raised, I haven't been able to recreate the issues you were seeing on the tables with that weird text overlay but I have fixed it so table cells don't wrap and use the ellipsis css rule so I think it should still be addressed. If not let me know and we can take a closer look.

For editing the crisis event notes I've left that for now as it's something I'd like to run it past Include UK.

phil-6 reviewed (changes requested):
Looking a lot better but still a few bugs when moving around the UI.  I really like the mobile design changes and the back button is a great addition. 
- After adding a note to a closed crisis event the back button appears to stop working. This actually appears to occur after adding any note.
- Back button also doesn't work after closing a crisis event.
- Simple formatting is not preserved in notes. (wrap outputs in simple_format() - https://apidock.com/rails/ActionView/Helpers/TextHelper/simple_format

It might be good hard code the back button paths. 

I can also no longer recreate the weird text overlay bug from before either. I suspect TurboLinks had a moment and hadn't loaded in correctly.

IeuanSkinner commented:
@phil-6 can you take another look at this? If happy I can merge this and then IJ-73

phil-6 reviewed (approved):
Looks good to me now.
